csp/tests/test_engine.py

```diff
@@ -841,6 +841,56 @@ def graph():
         with self.assertRaisesRegex(ValueError, "Dummy exception message"):
             csp.run(graph, starttime=datetime(2020, 1, 1), endtime=timedelta(seconds=1))
 
+    def test_adapter_manager_engine_shutdown_traceback(self):
+        from csp.impl.adaptermanager import AdapterManagerImpl, ManagedSimInputAdapter
+        from csp.impl.wiring import py_managed_adapter_def
+
+        d = {}
+
+        class TestAdapterManager:
+            def __init__(self):
+                self._impl = None
+
+            def subscribe(self):
+                return TestAdapter(self)
+
+            def _create(self, engine, memo):
+                self._impl = TestAdapterManagerImpl(engine)
+                return self._impl
+
+        class TestAdapterManagerImpl(AdapterManagerImpl):
+            def __init__(self, engine):
+                super().__init__(engine)
+
+            def start(self, starttime, endtime):
+                pass
+
+            def stop(self):
+                pass
+
+            def process_next_sim_timeslice(self, now):
+                try:
+                    [].pop()
+                except IndexError as e:
+                    d['tb'] = traceback.format_exc()
```

Comment:
We need to check the traceback after `csp.run`, the traceback info here doesn't necessarily mean its the same traceback info after `shutdown_engine` runs.

Also please just add this logic to the existing test, no need to make a second TestAdapterManager.